App/PyImports/QtInterface.py

```diff
@@ -78,13 +78,26 @@ def setExperimentDefinition(self, experiment_path: str):
         Parse the relevant phases file and update the corresponding model
         """
         CalculatorInterface.setExperimentDefinition(self, experiment_path)
+        self.updateCalculations()
+        self.projectDictChanged.emit()
+
+    def setExperimentDefinitionFromString(self, exp_cif_string: str):
+        """
+        Parse the relevant phases file and update the corresponding model
+        """
+        CalculatorInterface.setExperimentDefinitionFromString(self, exp_cif_string)
+        # NOTE THAT PbSO4 IS HARD CODDED. THIS SUCKS.
+        self.calculator.disassociatePhaseFromExp('pd', 'PbSO4')
+        self.calculator.associatePhaseToExp('pd', self.phasesIds()[0], 1)
```

Comment:
If proper phase name is implemented, as proposed in one of the previous comments, this hard coded name `Fe3O4` can be replaced with `phasesIds()[0]`

Reply:
With the right phase name used, is it even necessary to disassociate and re-associate?

Reply:
Simon confirmed the lines are now unnecessary and it was checked the application still loads xye data correctly.

Reply:
Yes, seems to work without disassociation.